src/main/java/com/example/bookstorewebapp/service/impl/BookServiceImpl.java

```diff
@@ -37,4 +37,18 @@ public BookDto findById(Long id) {
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        if (bookRepository.existsById(id)) {
+            Book book = bookMapper.toModel(requestDto);
+            book.setId(id);
+            return bookMapper.toDto(bookRepository.save(book));
+        }
+        throw new EntityNotFoundException("There is no book to update with id: " + id);
```

Comment:
let's not make the 'default' flow of the update method were it throws an exception. flip this `if` so that validation checks are structured like this:
```java
if (somethingWrong) {
    throw ex;
}
// another check
// business logic and other 'normal' flow of the program
```
this way you can stack up validations one **after** another and not one **inside** another, this is also called a [return early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8)